runtime/heiko/src/lib.rs

```diff
@@ -649,10 +649,30 @@ impl DataProvider<CurrencyId, TimeStampedPrice> for AggregatedDataProvider {
     }
 }
 
+impl DataProviderExtended<CurrencyId, TimeStampedPrice> for AggregatedDataProvider {
+    fn get_no_op(key: &CurrencyId) -> Option<TimeStampedPrice> {
+        Oracle::get_no_op(key)
+    }
+    #[allow(clippy::complexity)]
+    fn get_all_values() -> Vec<(CurrencyId, Option<TimeStampedPrice>)> {
+        Oracle::get_all_values()
+    }
+}
+
 impl pallet_prices::Config for Runtime {
     type Event = Event;
     type Source = AggregatedDataProvider;
     type FeederOrigin = EnsureRoot<AccountId>;
+    type StakingCurrency = StakingCurrency;
+    type LiquidCurrency = LiquidCurrency;
+    type LiquidStakingExchangeRateProvider = LiquidStakingExchangeRateProvider;
+}
+
+pub struct LiquidStakingExchangeRateProvider;
+impl ExchangeRateProvider for LiquidStakingExchangeRateProvider {
```

Comment:
I think that injects a single struct here is more decoupling and effective instead of a ‘big’ pallet, because `LiquidStaking` contains more than an exchange rate. how do you think of it? :)